tests/test_tracer.py

```diff
@@ -588,3 +589,36 @@ def test_tracer_dogstatsd_url():
     with pytest.raises(ValueError) as e:
         t = ddtrace.Tracer(dogstatsd_url='foo://foobar:12')
         assert str(e) == 'Unknown url format for `foo://foobar:12`'
+
+
+def test_tracer_fork():
+    t = ddtrace.Tracer()
+    original_pid = t._pid
+    original_writer = t.writer
+
+    @contextlib.contextmanager
+    def capture_failures(errors):
+        try:
+            yield
+        except AssertionError as e:
+            errors.put(e)
+
+    def task(t, errors):
+        # Start a new span to trigger process checking
+        t.trace('test', service='test')
+
+        # Assert we recreated the writer and have a new queue
+        with capture_failures(errors):
+            assert t._pid != original_pid
+            assert t.writer != original_writer
+            assert t.writer._trace_queue != original_writer._trace_queue
```

Comment:
can you check that we have the span we created L608 in the queue?

Reply:
yeah, I was trying to think about how to do that, but quick enough where the background worker won't pull it from the queue before we get a chance to

I could always manually stop the background thread first... ?